### ISSUE TITLE
Hook position not inserted in ps_hook table by default

### ISSUE THREAD

**Issue by idnovate** [PrestaShop/PrestaShop#15594]
**Describe the bug**
There's hook `displayCheckoutSummaryTop` at `/themes/classic/templates/checkout/_partials/cart-summary.tpl` template.

This hook is not in `ps_hook` table, so it doesn't appear in the dropdown at `Design -> Positions -> Transplant a module` and therefore can not be inserted.

**To Reproduce**
1. Add the following function to a module:

```
public function hookDisplayCheckoutSummaryTop()
{
    return true;
}
```
2. Go to `Design -> Positions -> Transplant a module`
3. Select module and check `Available hooks`. `displayCheckoutSummaryTop` is not displayed.

**Additional information**
PrestaShop version: 1.7.6.1

**Comment by idnovate** [comment]
**Fix**
Execute the following query:

```
`INSERT INTO `ps_hook` (`name`, `title`, `description`, `position`)
VALUES ('displayCheckoutSummaryTop', '', NULL, '1');`
```

Where could this be added to be executed in the upgrading scripts?

**Comment by khouloudbelguith** [comment]
Hi @idnovate,

I did not manage to reproduce the issue with PS1.7.6.1.
```
public function hookDisplayCheckoutSummaryTop()
{
    return true;
}
```
I just added this function into a module & I reset it.
![image](https://user-images.githubusercontent.com/16067358/65137816-e565fe80-da01-11e9-9112-18540fe993c7.png)

Thanks to check & feedback.

**Comment by idnovate** [comment]
Have you used `registerHook('displayCheckoutSummaryTop')`?

**Comment by khouloudbelguith** [comment]
@idnovate, yes!

**Comment by idnovate** [comment]
Hehe this function inserts the hook when it doesn't exist :) Please just add `function hookDisplayCheckoutSummaryTop()` to a new or existing module and don't register it on install.

**Comment by khouloudbelguith** [comment]
@idnovate,  the module does not do much. In order to display something on the front office, we have to add support for a few hooks. This is done by implementing the hooks’ methods, and that was actually done in the `install()` method we wrote earlier, using the `registerHook()` method.

Thanks!

**Comment by idnovate** [comment]
This is not how it works. It's inconsistent. Test the following:

1. Add both functions to a module:
```
public function hookDisplayCheckoutSummaryTop()
{
    return true;
}

public function hookDisplayCartExtraProductActions()
{
    return true;
}
```
2. Go to `Design -> Positions -> Transplant a module`
3. Check that you can transplant `DisplayCartExtraProductActions` but no `DisplayCheckoutSummaryTop`.

The dropdown check which hooks are defined in the module and display them in the dropdown. This has been working this way always, there wasn't the need to register the hook in the install to transplant a module into a hook. 

**The problem is that `DisplayCheckoutSummaryTop` is not in `ps_hook` table.**

**Comment by khouloudbelguith** [comment]
@idnovate, I tried with those two functions & those two hooks are displayed.
![image](https://user-images.githubusercontent.com/16067358/65143643-68d91d00-da0d-11e9-8b56-de9d108a0977.png)

Thanks!

**Comment by idnovate** [comment]
1. Install this module
[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/3626123/dummymodule.zip)
2. Go to `Design -> Positions -> Transplant a module`
3. Select `dummymodule`
4. Check available hooks. It should appear both `DisplayCheckoutSummaryTop` and `DisplayCartExtraProductActions` hooks but it doesn't. `DisplayCheckoutSummaryTop` is not displayed.

![image](https://user-images.githubusercontent.com/15104724/65145121-10a81800-da19-11e9-8eb7-465505ebe0df.png)

Please test it in a different store, as you have previously registered hook `DisplayCheckoutSummaryTop` so it's in your `ps_hook` table and you won't be able to reproduce it.

**Comment by khouloudbelguith** [comment]
@idnovate, thanks for this clarifications.
I manage to reproduce the issue with PS1.7.6.0 & PS1.7.5.2
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by idnovate** [comment]
Actually the fix is this: https://github.com/PrestaShop/PrestaShop/issues/15594#issuecomment-532606917

**Comment by Progi1984** [comment]
Fixed by @okom3pom in PrestaShop/autoupgrade/pull/422 & tested by @marwachelly
